### PR TITLE
Add a lil README

### PR DESCRIPTION
**Pull request by stephencelis**
And fix some UIKit stuff

### CODE REVIEW

File: Sources/SnapshotAssertion/SnapshotAssertion.swift [hunk 1]
```diff
@@ -4,8 +4,8 @@ var diffTool: String? = nil
 
 public protocol Diffable: Equatable {
   static var diffableFileExtension: String? { get }
+  static func fromDiffableData(_ data: Data) -> Self
```
Comment (stephencelis) [inline Sources/SnapshotAssertion/SnapshotAssertion.swift]: Non-final classes (like `UIImage`) have trouble conforming to protocol-based `convenience init`s.

**Comment by stephencelis** [comment]
Should flesh the README out a _lot_ more (with images showing off the `XCTAttachment` integration, etc.), but this is a nice start.

File: Sources/SnapshotAssertion/Cocoa.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import Cocoa
```
Comment (stephencelis) [inline Sources/SnapshotAssertion/Cocoa.swift]: Snuck some preliminary support for Cocoa stuff here. Means we can maybe render some `NSWebView` stuff more quickly for `Html`/`Css`.

File: Sources/SnapshotAssertion/Cocoa.swift [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public var diffableData: Data {
+      let cgImage = self.cgImage(forProposedRect: nil, context: nil, hints: nil)!
+      let rep = NSBitmapImageRep(cgImage: cgImage)
+      rep.size = self.size
+      let data = rep.representation(using: .png, properties: [:])!
+      return data
+    }
+
+    public func diff(comparing other: Data) -> XCTAttachment? {
+      return nil
```
Comment (stephencelis) [inline Sources/SnapshotAssertion/Cocoa.swift]: TODO

**Comment by stephencelis** [comment]
Not too shabby:

screen shot 2017-07-09 at 1 17 23 am

File: Sources/SnapshotAssertion/Cocoa.swift (hunk 2 above)
Comment (mbrandonw) [inline Sources/SnapshotAssertion/Cocoa.swift]: yes! but we'd use `WKWebView`! it's available for both macOS AND ios!

File: Sources/SnapshotAssertion/SnapshotAssertion.swift (hunk 1 above)
Comment (mbrandonw) [inline Sources/SnapshotAssertion/SnapshotAssertion.swift]: yuck

File: Tests/SnapshotAssertionTests/SnapshotAssertionTests.swift [hunk 4]
```diff
@@ -2,15 +2,27 @@ import XCTest
...
+    }
+  #endif
+
+  #if os(macOS)
+    func testCocoa() {
+      let button = NSButton()
+      button.bezelStyle = .rounded
+      button.title = "Push Me"
+      button.sizeToFit()
+      assertSnapshot(matching: button)
```
Comment (mbrandonw) [inline Tests/SnapshotAssertionTests/SnapshotAssertionTests.swift]: ok that's cool!

File: Sources/SnapshotAssertion/Cocoa.swift (hunk 2 above)
Comment (stephencelis) [inline Sources/SnapshotAssertion/Cocoa.swift]: See #12 :(

**Pull request merged**